gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java

```diff
@@ -118,7 +117,19 @@ public Blob(Storage storage, BlobInfo info) {
    */
   public Blob(Storage storage, String bucket, String blob) {
     this.storage = checkNotNull(storage);
-    this.info = BlobInfo.of(checkNotNull(bucket), checkNotNull(blob));
+    this.info = BlobInfo.builder(BlobId.of(bucket, blob)).build();
```

Comment:
Ok, if you prefer we can leave it like this for now and I will open and issue for us to discuss it.

Here is my reason (will copy it to the issue):
Considering the following operations
info, exists, content, update, delete, copyTo, reader, writer and signUrl

In all cases info matters if options are used (I will create a seperate issue for adding the options to the  `exists` method) and in many cases info matters even without the options (e.g. info, copyTo). Considering our example, I think this will make API cleaner (and will not require
the explicit blob = blob.reload()).

The reasons I think we don't want to force-get when BlobInfo is provided are:
- we don't want to double-reload when our API is returning the `Blob`. I guess we can get around that with a "special" constructor...
- That is the way for the user to force a new Info (e.g. for write).

Now when we clearly separate between blob identity and blob metadata, I think it is not
that unexpected that if one supply the metadata we should respect it as is (though documentation should make it clear).